### PR TITLE
Added table about message correlation

### PR DESCRIPTION
**Pull request by jonathanlukas**
## Description

The new section contains a table that explains message behavior as the publishMessage command is funny.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by christinaausley** [comment]
Is this only for the `next` version of documentation? I noticed this is already available but undocumented -- does this belong in any versioned docs as well?

File: docs/components/concepts/messages.md
Comment (koevskinikola) [inline docs/components/concepts/messages.md:101]: :wrench: Typo suggestion:
```suggestion
| empty string    | set        | set < 0      | Intermediate event | The message is correlated during the lifetime of the message if a matching subscription to the empty string is active and there is no equal message in the buffer.           |
```

**Review by koevskinikola — CHANGES_REQUESTED** [review]
Thanks for adding this @jonathanlukas!

The table looks pretty good. I added some feedback and some things to clarify before I approve the changes.

File: docs/components/concepts/messages.md
Comment (jonathanlukas) [reply docs/components/concepts/messages.md:1]: I added a link to each occurence that points to the paragraph _Message uniqueness_ which explains it

**Review by christinaausley — APPROVED** [review]

**Comment by jonathanlukas** [comment]
Oh yes @christinaausley . I will would like to wait for final approval and then replicate the new content to old versions of this page (including all until 8.1).

**Review by koevskinikola — DISMISSED** [review]
LGTM @jonathanlukas :tada:

**Review by christinaausley — APPROVED** [review]

**Pull request merged**
